File: lc2462/Solution.java
import java.util.PriorityQueue;

class Solution {

    public static long totalCost(int[] costs, int k, int candidates) {

        PriorityQueue<Integer> queue1 = new PriorityQueue<>();
        PriorityQueue<Integer> queue2 = new PriorityQueue<>();

        long totalCost = 0L;
        int i = 0, j = costs.length - 1;

        while (k != 0) {
            while (queue1.size() < candidates && i <= j) {
                queue1.add(costs[i]);
                i++;
            }
            while (queue2.size() < candidates && j >= i) {
                queue2.add(costs[j]);
                j--;
            }

            int leftMin = queue1.peek() != null ? queue1.peek() : Integer.MAX_VALUE;
            int rightMin = queue2.peek() != null ? queue2.peek() : Integer.MAX_VALUE;

            if (leftMin <= rightMin) {
                totalCost += leftMin;
                queue1.poll();
            } else {
                totalCost += rightMin;
                queue2.poll();
            }

            k--;
        }

        return totalCost;
    }
}

File: lc2462/solution.py
import heapq
from typing import List

class Solution:
    def totalCost(self, costs: List[int], k: int, candidates: int) -> int:
        
        queue1, queue2 = [], []
        total_cost, i, j = 0, 0,len(costs) - 1

        while k:
            while len(queue1) < candidates and i <= j:
                heapq.heappush(queue1, costs[i])
                i += 1
            while len(queue2) < candidates and j >= i:
                heapq.heappush(queue2, costs[j])
                j -= 1
            
            left_min = queue1[0] if queue1 else float('inf')
            right_min = queue2[0] if queue2 else float('inf')

            if left_min <= right_min:
                total_cost += left_min
                heapq.heappop(queue1)
            else:
                total_cost += right_min
                heapq.heappop(queue2)
            
            k -= 1

        return total_cost
